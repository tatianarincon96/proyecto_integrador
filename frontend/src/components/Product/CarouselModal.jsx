import "./carouselModal.css"
import "react-responsive-carousel/lib/styles/carousel.min.css";
//import { Modal } from 'react-responsive-modal'
import { Carousel } from "react-responsive-carousel"
import { Modal, ModalGateway } from "react-images"


export default function CarouselModal( { images, viewerIsOpen, setCurrentImage, setViewerIsOpen } ) {

    const closeLightbox = () => {
        setCurrentImage(0);
        setViewerIsOpen(false);
    };

    return (
        <div>
            <ModalGateway>
                {viewerIsOpen ? (
                    <Modal onClose={closeLightbox}>
                        <div>
                            <div className={`closeButton`} onClick={closeLightbox}>X</div>
                            <Carousel
                                statusFormatter={(current, total) => `${current}/${total}`}
                                autoPlay={true}
                                showIndicators={false}
                                interval={3000}
                                thumbWidth={148}
                                width="44vw"
                                stopOnHover={true}
                                emulateTouch={true}
                                infiniteLoop={true}
                                useKeyboardArrows={true}
                            >
                                {images.map((image) => {
                                    return (<div>
                                        <img src={image.url} />
                                    </div>)
                                })}

                            </Carousel>
                        </div>
                    </Modal>
                ) : null}
            </ModalGateway>
        </div>

    );
}

