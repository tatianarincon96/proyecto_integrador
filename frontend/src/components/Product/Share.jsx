import { Modal, ModalGateway } from "react-images";
import Styles from "./styles.module.css";
import facebook from "./icons/facebook.svg";
import twitter from "./icons/twitter.svg";
import whatsapp from "./icons/whatsapp.svg";
import email from "./icons/email.png";

export default function Share(props) {

    const closeShare = () => {
        props.setShareIsOpen(false);
    };
    return (
        <div>
            <ModalGateway>
                {props.shareIsOpen ? (
                    <Modal onClose={closeShare}>
                        <div className={Styles.containerSharePrincipal}>
                            
                            <h3>Compartir</h3>
                            <div className={Styles.containerShare}>
                                <div className={Styles.shareBlock}>
                                    <div className={Styles.shareItem}>
                                        <a href={`mailto:?Subject=Mira este alojamiento localhost:3000/product/${props.id}`} target="_blank" ><img src={email} alt="logo email" /></a>
                                        <p>Correo</p>
                                    </div>
                                    <div className={Styles.shareItem}>
                                        <a href={`https://www.facebook.com/sharer/sharer.php?u=[localhost:3000/product/${props.id}]`} target="_blank"><img src={facebook} alt="logo facebook" /></a>
                                        <p>Facebook</p>
                                    </div>
                                </div>
                                <div className={Styles.shareBlock}>
                                    <div className={Styles.shareItem}>
                                        <a href={`https://twitter.com/intent/tweet?text=[Mira este producto ]&url=[localhost:3000/product/${props.id}]&hashtags=[GranAlojamiento]`} target="_blank"><img src={twitter} alt="logo twitter" /></a>
                                        <p>Twitter</p>
                                    </div>
                                    <div className={Styles.shareItem}>
                                        <a href={`https://api.whatsapp.com/send?text=[Mira este alojamiento: localhost:3000/product/${props.id}]`} target="_blank"><img src={whatsapp} alt="logo whatsapp" /></a>
                                        <p>Whatsapp</p>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </Modal>) : null}
            </ModalGateway>
        </div>
    )
}