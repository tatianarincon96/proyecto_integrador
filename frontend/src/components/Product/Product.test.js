import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import CarouselModal from "./CarouselModal";
import Share from "./Share";

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe("Probando el componente <CarouselModal/>", () => {
    let wrapper;
    let props={
        images:[{url:""}],
        viewerIsOpen:true,
        setCurrentImage: jest.fn(),
        setViewerIsOpen: jest.fn()
    }
   
    beforeEach(() => {
        wrapper = shallow(<CarouselModal/>)
    });

    it("Deberia mostrar <CarouselModal/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia cerrarse el carousel al hacer click en la cruz", ()=>{
        wrapper = shallow(<CarouselModal {...props}/>)
        wrapper.find("div.closeButton").simulate("close")
        expect(wrapper.prop("viewerIsOpen")).toBeFalsy()
    })
})

describe("Probando el componente <CarouselModal/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<Share/>)
    });

    it("Deberia mostrar <Share/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})
