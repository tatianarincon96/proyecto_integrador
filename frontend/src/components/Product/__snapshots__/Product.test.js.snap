// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <CarouselModal/> Deberia mostrar <CarouselModal/> correctamente 1`] = `
<div>
  <ModalGateway
    target={<body />}
  />
</div>
`;

exports[`Probando el componente <CarouselModal/> Deberia mostrar <Share/> correctamente 1`] = `
<div>
  <ModalGateway
    target={<body />}
  />
</div>
`;
